### ISSUE TITLE
Push Hyku fixes back into pals

### ISSUE THREAD

**Issue by bkiahstroud** [scientist-softserv/palni-palci#382]
## Summary

While fixing specs in #369, a handful of bugs were found. These are being fixed as part of https://github.com/samvera/hyku/pull/1859. 

Once that PR is merged, the fixes contained within it need to be backported into the pals repo. 

## Acceptance Criteria

- [x] Groups with Roles related bug fixes have been pushed back into pals from Hyku
  - [x] Code changes have been brought into pals (`cherry-pick`ed) 
  - [x] Changes in diff have been reviewed and verified (except specs)
  - [x] Make any additional changes necessary (bringing overrides, configs, etc. inline with updates) 
  - [x] Ensure new decorator overrides merge with existing `class_eval` overrides 
  - [x] Make sure no custom pals code has been lost 
  - [x] Verify translations 
  - [x] Fix failing specs 
  - [ ] ~~Fix rubocop~~ (split to #418) 
  - [ ] ~~Skip labeled tests in CI~~ (split to #417) 

## Testing Instructions 

1. Navigate to an existing tenant 
2. Sign in as an admin 
3. Navigate to Dashboard > Collections > All Collections. Look through each Admin Set until you find one with a mediated deposit Workflow. If none exist, create a new Admin Set and give it a mediated deposit workflow 
4. Navigate to Works. Click the Create New Work button and select any work type 
5. Make the work Public. Fill out the required fields 
6. On the Relationships tab, select the Admin Set with a mediated deposit workflow (see Step 4) 
7. Click Save 
8. Navigate to Dashboard > Review Submissions 
9. Verify your new work shows up in the Under Review tab 
10. Click your work's title 
11. In the Review and Approval dropdown, select Approve. Click Submit 
12. Verify the work moves into the Deposited state 
13. In a private/incognito tab, search for your work's title. Verify you can see the work 
14. Repeat steps 2-13 in a newly created tenant

**Comment by bkiahstroud** [comment]
Got [this error](https://scientist-inc.sentry.io/issues/3623612603/?query=is%3Aunresolved&referrer=issue-stream) when approving a Public `GenericWork` on a new tenant (deploy)

**Comment by DiemBTran** [comment]
**This passes SoftServ QA:** See attached screen recording. **QA Note:** step 9: `Verify your new work shows up in the Under Review tab` is bugged for **pre-existing tenants**. This will be addressed in a new ticket: (will add ticket number once I create it). For now, you may have to approve the work from its work show page (without navigating to `Dashboard` > `Review Submissions`

tested on:
- existing tenant
     - `Review Submissions` tab: https://qa-9.commons-archive.org/admin/workflows?locale=en#under-review
     - public work: https://qa-9.commons-archive.org/concern/images/fa172583-166f-43d4-9434-73277d7410f4?locale=en
- new tenant
     - `Review Submissions` tab: https://2023-diem-new-tenant.commons-archive.org/admin/workflows?locale=en#under-review
     - public work: https://2023-diem-new-tenant.commons-archive.org/concern/etds/9d5fc523-f806-45a2-b5de-00b4377ac106?locale=en

<details><summary>Works created with a mediated workflow can be approved and deposited. When they are pending review, they cannot be viewed. When they are approved and deposited, they can be viewed.</summary>

https://user-images.githubusercontent.com/29311858/219489821-7a65e65e-55fd-48e7-8467-54b31b6c8a9c.mp4
</details>
